src/pages/dashboardOverview/components/activeProductsSection/components/ActiveProductCard.tsx

```diff
@@ -1,54 +1,118 @@
-import { Card, Grid, Box, Typography } from '@mui/material';
-import { formatDateAsLongString } from '@pagopa/selfcare-common-frontend/utils/utils';
-import { useTranslation } from 'react-i18next';
-import { useTokenExchange } from '../../../../../hooks/useTokenExchange';
+import React, { useMemo } from 'react';
+import { Typography, Button, Box, Grid, Card, CardContent, Link, Chip } from '@mui/material';
+import { Trans } from 'react-i18next';
+import { useUnloadEventOnExit } from '@pagopa/selfcare-common-frontend';
+import { useHistory } from 'react-router';
+import { resolvePathVariables } from '@pagopa/selfcare-common-frontend/utils/routes-utils';
+import { ENV } from '../../../../../utils/env';
 import { Party } from '../../../../../model/Party';
-import { Product } from '../../../../../model/Product';
-import BaseProductCard from '../../productCard/BaseProductCard';
+import { Product, SubProduct } from '../../../../../model/Product';
 
 type Props = {
+  cardTitle: string;
+  buttonLabel: string;
+  disableBtn: boolean;
+  urlLogo?: string;
+  btnAction?: () => void;
+  tooltip: string;
   party: Party;
   product: Product;
 };
+export default function ActiveProductCard({
+  cardTitle,
+  buttonLabel,
+  disableBtn,
+  urlLogo,
+  btnAction,
+  party,
+  product,
+}: Props) {
+  const onExit = useUnloadEventOnExit();
+  const history = useHistory();
+  const usersRoute = ENV.ROUTES.USERS;
+  const usersPath = resolvePathVariables(usersRoute, {
+    partyId: party.partyId,
+  });
 
-export default function ActiveProductCard({ party, product }: Props) {
-  const { t } = useTranslation();
-  const { invokeProductBo } = useTokenExchange();
-  const isDisabled = product.authorized === false;
-  const lastServiceActivationDate = undefined; // actually this info is not available
-
+  const activeSubProducts: Array<SubProduct> = useMemo(
+    () => product.subProducts.filter((p) => p.status === 'ACTIVE' && !disableBtn) ?? [],
```

Comment:
why this disableBtn filter condition?

Reply:
ok, thus don't check this values for each element of the array, check once for all